subsys/bluetooth/controller/Kconfig

```diff
@@ -536,18 +542,21 @@ config BT_CTLR_ADV_PERIODIC_ADI_SUPPORT
 	  Enable support for adding AdvDataInfo in Periodic Advertising PDUs.
 
 config BT_CTLR_SYNC_PERIODIC
-	bool "LE Periodic Advertising in Synchronization State" if !BT_LL_SW_SPLIT
+	prompt "LE Periodic Advertising in Synchronization State" if !BT_LL_SW_SPLIT
```

Comment:
@tejlmand , I think you tried this with one of the older commits pushed to this PR. I'm not able to reproduce what you got. This is what I did:
```
cd <workspace>/nrf 
git checkout 7371dec6308cea591bc1b450f1710f0c73b60b87
west config manifest.path nrf
west update
cd <workspace>/zephyr
git cherry-pick a82f92995207e5101eacb365710f963f1d700c90
```
With BT_LL_SW_SPLIT:
```
$ west build -b nrf52840dk_nrf52840 samples/bluetooth/periodic_adv/ -- -DCONFIG_BT_LL_SW_SPLIT=y -DCONFIG_BT_CTLR_SYNC_PERIODIC=y
-- west build: generating a build system
Including boilerplate (Zephyr base): C:/ncs/zephyr/cmake/app/boilerplate.cmake
-- Application: C:/ncs/zephyr/samples/bluetooth/periodic_adv
-- Zephyr version: 2.7.99 (C:/ncs/zephyr), build: v2.7.99-ncs1-rc1-69-gc9a09ec4fc13
-- Found Python3: C:/Python36/python3.exe (found suitable exact version "3.6.8") found components: Interpreter
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: nrf52840dk_nrf52840
-- Cache files will be written to: C:\Users\ruge\AppData\Local/.cache/zephyr
-- Found dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.4.7", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (C:/Program Files (x86)/GNU Tools ARM Embedded/8 2019-q3-update)
-- Found BOARD.dts: C:/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts
-- Generated zephyr.dts: C:/ncs/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: C:/ncs/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: C:/ncs/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: C:/ncs/zephyr/build/zephyr/dts.cmake

warning: Experimental symbol BT_LL_SW_SPLIT is enabled.


warning: Experimental symbol BT_CTLR_ADV_EXT is enabled.


warning: Experimental symbol BT_CTLR_ADV_PERIODIC is enabled.


warning: Experimental symbol BT_CTLR_SYNC_PERIODIC is enabled.

Parsing C:/ncs/zephyr/Kconfig
Loaded configuration 'C:/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
Merged configuration 'C:/ncs/zephyr/samples/bluetooth/periodic_adv/prj.conf'
Merged configuration 'C:/ncs/zephyr/samples/bluetooth/periodic_adv/boards/nrf52840dk_nrf52840.conf'
Merged configuration 'C:/ncs/zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf'
Configuration saved to 'C:/ncs/zephyr/build/zephyr/.config'
Kconfig header saved to 'C:/ncs/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 8.3.1
-- The CXX compiler identification is GNU 8.3.1
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files (x86)/GNU Tools ARM Embedded/8 2019-q3-update/bin/arm-none-eabi-gcc.exe
CMake Warning at C:/ncs/zephyr/subsys/bluetooth/controller/CMakeLists.txt:29 (message):


  CONFIG_BT_CTLR_ADV_EXT=y, Advertising Extensions Feature in Zephyr
  Bluetooth Controller is EXPERIMENTAL.


-- Configuring done
-- Generating done
-- Build files have been written to: C:/ncs/zephyr/build
-- west build: building application
[1/225] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[2/225] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
```
With BT_LL_SOFTDEVICE:
```
$ west build -b nrf52840dk_nrf52840 samples/bluetooth/periodic_adv/ -- -DCONFIG_BT_LL_SOFTDEVICE=y -DCONFIG_BT_CTLR_SYNC_PERIODIC=y
-- west build: generating a build system
Including boilerplate (Zephyr base): C:/ncs/zephyr/cmake/app/boilerplate.cmake
-- Application: C:/ncs/zephyr/samples/bluetooth/periodic_adv
-- Zephyr version: 2.7.99 (C:/ncs/zephyr), build: v2.7.99-ncs1-rc1-69-gc9a09ec4fc13
-- Found Python3: C:/Python36/python3.exe (found suitable exact version "3.6.8") found components: Interpreter
-- Found west (found suitable version "0.12.0", minimum required is "0.7.1")
-- Board: nrf52840dk_nrf52840
-- Cache files will be written to: C:\Users\ruge\AppData\Local/.cache/zephyr
-- Found dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.4.7", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (C:/Program Files (x86)/GNU Tools ARM Embedded/8 2019-q3-update)
-- Found BOARD.dts: C:/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840.dts
-- Generated zephyr.dts: C:/ncs/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: C:/ncs/zephyr/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: C:/ncs/zephyr/build/zephyr/include/generated/device_extern.h
-- Including generated dts.cmake file: C:/ncs/zephyr/build/zephyr/dts.cmake
Parsing C:/ncs/zephyr/Kconfig
Loaded configuration 'C:/ncs/zephyr/boards/arm/nrf52840dk_nrf52840/nrf52840dk_nrf52840_defconfig'
Merged configuration 'C:/ncs/zephyr/samples/bluetooth/periodic_adv/prj.conf'
Merged configuration 'C:/ncs/zephyr/samples/bluetooth/periodic_adv/boards/nrf52840dk_nrf52840.conf'
Merged configuration 'C:/ncs/zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf'
Configuration saved to 'C:/ncs/zephyr/build/zephyr/.config'
Kconfig header saved to 'C:/ncs/zephyr/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 8.3.1
-- The CXX compiler identification is GNU 8.3.1
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files (x86)/GNU Tools ARM Embedded/8 2019-q3-update/bin/arm-none-eabi-gcc.exe
-- Configuring done
-- Generating done
-- Build files have been written to: C:/ncs/zephyr/build
-- west build: building application
[1/202] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
```

Reply:
@rugeGerritsen you're proving my point.

You are **supposed** to get an error.
I wrote:
> then I can suddenly set this variable.

User's are **not** supposed to be able to enable that setting, but with your changes they can.
Try the same on `ncs/main` and you'll see that very same build command will report an error to the user.

Reply:
@tejlmand , sorry, I don't understand what you are trying to point out. Users **are** supposed to be able to set `BT_CTLR_SYNC_PERIODIC`. 

The logs I posted above are from ncs/main (otherwise I would not be able to use CONFIG_BT_LL_SOFTDEVICE)

Reply:
No, as soon as you do this:
```
cd <workspace>/zephyr
git cherry-pick a82f92995207e5101eacb365710f963f1d700c90
```

then you have your changes on top

Reply:
@tejlmand , sorry for my lack of understanding. Can you briefly try to point out:
1. What are you trying to achieve. 
2. What is the expected output
3. What do you actually see.

I don't understand what you are trying to say with
> No, as soon as you do this:
> cd <workspace>/zephyr
> git cherry-pick a82f92995207e5101eacb365710f963f1d700c90
> then you have your changes on top

Reply:
Hint, this indicates you are building with your cherry-pick commit:
> -- Zephyr version: 2.7.99 (C:/ncs/zephyr), build: v2.7.99-ncs1-rc1-69-gc9a09ec4fc13

Cause `c9a09ec4fc13` is not a valid SHA here https://github.com/nrfconnect/sdk-zephyr, hence it must be a SHA only available at your local workspace.

Reply:
> What are you trying to achieve.

I'm concerned that you accidentally are changing the dependencies of those symbols when tryin to solve an issue that is not there in the first place. See this message: https://github.com/zephyrproject-rtos/zephyr/pull/42319#issuecomment-1034759189

So i'm trying to avoid a PR going in, which suddenly allows users to enable settings they are not supposed to enable.

> What is the expected output

According to you commit message, this is the expected output:
>  Bluetooth: Controller: Only show one prompt for experimental configs
>
> When browsing through the tree, you will now no longer see two entries
> for each symbol that is experimental for BT_LL_SW_SPLIT
>
> The prompt message for experimental messages have now been rephrased to
> include (Split Link Layer)

and therefore I don't assume kconfig dependencies should change.

> What do you actually see.

Changed kconfig dependencies.